### PR TITLE
Trees: Add toString for AST nodes which are not case classes

### PR DESCRIPTION
**Pull request by smarter**
This makes debugging trees easier

Review by @odersky .

### CODE REVIEW

**Comment by adriaanm** [comment]
https://github.com/scala/scabot/blob/master/README.md#commands

**Comment by adriaanm** [comment]
The only time I'm aware of you need to trigger a synch is when Jenkins was restarted while scabot was fetching the job status. It normally checks all PRs but it keeps a cache so it can get out of synch

**Comment by odersky** [comment]
LGTM

File: src/dotty/tools/dotc/ast/Trees.scala [hunk 1]
```diff
@@ -346,7 +346,9 @@ object Trees {
   }
 
   class BackquotedIdent[-T >: Untyped] private[ast] (name: Name)
-    extends Ident[T](name)
+    extends Ident[T](name) {
+    override def toString = "BackquotedIdent($name)"
```
Comment (namin) [inline src/dotty/tools/dotc/ast/Trees.scala]: missing the s prefix for string interpolation

File: src/dotty/tools/dotc/ast/Trees.scala [hunk 2]
```diff
@@ -355,7 +357,9 @@ object Trees {
   }
 
   class SelectWithSig[-T >: Untyped] private[ast] (qualifier: Tree[T], name: Name, val sig: Signature)
-    extends Select[T](qualifier, name)
+    extends Select[T](qualifier, name) {
+    override def toString = "SelectWithSig($qualifier, $name, $sig)"
```
Comment (namin) [inline src/dotty/tools/dotc/ast/Trees.scala]: here too missing the s for string interpolation

File: src/dotty/tools/dotc/ast/Trees.scala (hunk 1 above)
Comment (smarter) [inline src/dotty/tools/dotc/ast/Trees.scala]: woops, thanks for catching that.

**Pull request merged**